src/agent/src/mount.rs

```diff
@@ -1012,6 +1015,40 @@ fn ensure_destination_file_exists(path: &Path) -> Result<()> {
     Ok(())
 }
 
+/// Parse mtab and return the device which is mounted at a given directory
+pub fn get_mount_device(mount_dir: &Path) -> Result<String> {
+    let dir = mount_dir.to_string_lossy().into_owned();
+    let f = File::open(MTAB_PATH)?;
+    let reader = BufReader::new(f);
+
+    for line in reader.lines() {
+        let tem_line = line?;
+        let parts: Vec<&str> = tem_line.split_whitespace().collect();
+        if parts.contains(&dir.as_str()) && !parts.is_empty() {
+            return Ok(String::from(parts[0]));
+        }
+    }
+
+    return Err(anyhow!("can't find mount point {:?}", dir));
+}
+
+#[instrument]
+pub fn expand_file_system(mountpoint: &String, logger: Logger) -> Result<()> {
+    let path = Path::new(mountpoint);
+    match get_mount_device(path) {
+        Ok(device_path) => {
```

Comment:
Since you are doing nothing in case of error, I think this will save a `match` statement:

```rust
let device_path = get_mount_device(path)?;
......
```